src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\node\Plugin\EntityReferenceSelection\OgSelection.
+ */
+
+namespace Drupal\og\Plugin\EntityReferenceSelection;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\Plugin\EntityReferenceSelection\DefaultSelection;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\og\Og;
+
+/**
+ * Provide default OG selection handler.
+ *
+ * @EntityReferenceSelection(
+ *   id = "default:og",
+ *   label = @Translation("OG selection"),
+ *   entity_types = {"node"},
+ *   group = "default",
+ *   weight = 1
+ * )
+ */
+class OgSelection extends DefaultSelection {
+
+  private $targetType;
+
+  /**
+   * Get the current account.
+   *
+   * @return AccountInterface
+   */
+  public function getAccount() {
+    if (empty($this->account)) {
+      $this->setAccount(\Drupal::currentUser()->getAccount());
+    }
+
+    return $this->currentUser;
+  }
+
+  /**
+   * Set the current object account.
+   *
+   * @param AccountInterface $account
+   */
+  public function setAccount(AccountInterface $account) {
+    $this->currentUser = $account;
+  }
+
+  /**
+   * Overrides the basic entity query object. Return only group in the matching
+   * results.
+   *
+   * @param string|null $match
+   *   (Optional) Text to match the label against. Defaults to NULL.
+   * @param string $match_operator
+   *   (Optional) The operation the matching should be done with. Defaults
+   *   to "CONTAINS".
+   *
+   * @return \Drupal\Core\Entity\Query\QueryInterface
+   *   The EntityQuery object with the basic conditions and sorting applied to
+   *   it.
+   */
+  protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS') {
+    $query = parent::buildEntityQuery($match, $match_operator);
+
+    $identifier_key = \Drupal::entityManager()->getDefinition($this->configuration['target_type'])->getKey('id');
+    $this->targetType = $this->configuration['target_type'];
+    $user_groups = $this->getUserGroups();
+    $bundles = Og::groupManager()->GetAllGroupBundles($this->configuration['target_type']);
```

Comment:
this is actually not totally right. We should take this info from the $field->handler_settings. If it doesn't limit bundles, then we can get them all from `getAllGroupBundles`